src/cli/CliRandomizer.java

```diff
@@ -0,0 +1,223 @@
+            fis.close();
+        } catch (UnsupportedOperationException ex) {
+            ex.printStackTrace();
+            System.err.println(ex.getMessage());
+            return null;
+        } catch (IllegalArgumentException ex) {
+            ex.printStackTrace();
+            System.err.println(bundle.getString("GUI.invalidSettingsFile") + "%n");
+            return null;
+        } catch (IOException ex) {
+            ex.printStackTrace();
+            System.err.println(bundle.getString("GUI.settingsLoadFailed") + "%n");
+            return null;
+        }
+        return settings;
+    }
+
+    private static boolean canSaveAsDirectory(String romFilePath) {
+        File romFileHandler = new File(romFilePath);
+        RomHandler romHandler = CliRandomizer.getRomHandler(romFilePath);
+        assert romHandler != null;
+        romHandler.loadRom(romFileHandler.getAbsolutePath());
+        return romHandler.generationOfPokemon() == 6 || romHandler.generationOfPokemon() == 7;
+    }
+
+    private static boolean isExtensionValid(String romFilePath) {
+        File romFileHandler = new File(romFilePath);
+        RomHandler romHandler = CliRandomizer.getRomHandler(romFilePath);
+        List<String> extensions = new ArrayList<>(Arrays.asList("sgb", "gbc", "gba", "nds", "cxi"));
+        assert romHandler != null;
+        extensions.remove(romHandler.getDefaultExtension());
+        File fh = FileFunctions.fixFilename(romFileHandler, romHandler.getDefaultExtension(), extensions);
+        if (romHandler instanceof AbstractDSRomHandler || romHandler instanceof Abstract3DSRomHandler) {
+            String currentFN = romHandler.loadedFilename();
+            return !currentFN.equals(fh.getAbsolutePath());
+        }
+        return true;
+    }
+
+    private static boolean performDirectRandomization(
+            String settingsFilePath,
+            String sourceRomFilePath,
+            String destinationRomFilePath
+    ) {
+        return CliRandomizer.performDirectRandomization(settingsFilePath, sourceRomFilePath, destinationRomFilePath, false);
+    }
+
+    private static boolean performDirectRandomization(
+            String settingsFilePath,
+            String sourceRomFilePath,
+            String destinationRomFilePath,
+            boolean saveAsDirectory
+    ) {
+        Settings settings = CliRandomizer.getValidatedSettings(settingsFilePath);
+        if (settings == null) {
+            return false;
+        }
+
+        try {
+            File romFileHandler = new File(sourceRomFilePath);
+            RomHandler romHandler = CliRandomizer.getRomHandler(sourceRomFilePath);
+            if (romHandler == null) {
+                System.err.printf(bundle.getString("GUI.unsupportedRom") + "%n", romFileHandler.getName());
+                return false;
+            }
+
+            romHandler.loadRom(sourceRomFilePath);
+            CliRandomizer.displaySettingsWarnings(settings, romHandler);
```

Comment:
@Ajarmar I don't love how some of these helper methods take the `RomHandler` directly and some take a ROM filepath and instantiate the ROMHandler from it but that's kind of required without the filepath/handler exposed from the ROMHandler itself. If there's no problem creating a public method for `RomHandler` that will return the file path then we could make these method signatures more consistent. It would be nice to be able to pass just the `RomHandler` since some methods may require `loadRom` to have been called and that can get very repetitive if we're doing that in every helper method.

Reply:
To be honest - I don't think the helper methods help much here, I understand the desire to split things up into separate methods but the repeated creations of `File` and `RomHandler` objects just make the code more confusing to read. Your previous commit looked mostly fine; if you can go back to that one and push it, I have a few other things I'd like to give feedback on as well.

Reply:
@Ajarmar Fair enough, most recent commit reverted